source/core/src/core/utility/delta_table_helper.py

```diff
@@ -54,13 +55,42 @@ def append_if_not_exists(
     ).whenNotMatchedInsertAll().execute()
 
 
-# Get target filters
 def get_target_filter_for_datetime_clustering_key(
     update_df: DataFrame, clustering_col: str, current_alias_table_name: str
 ) -> str:
-    dates_to_filter = [row[0] for row in update_df.select(col(clustering_col).cast("date")).distinct().collect()]
+    # We get and sort all dates for the clustering key in our input.
+    dates_to_filter = sorted(
+        [row[0] for row in update_df.select(col(clustering_col).cast("date")).distinct().collect()]
+    )
     if len(dates_to_filter) == 0:
         return "TRUE"
 
-    joined_string = ",".join([str(date) for date in dates_to_filter])
-    return f"CAST({current_alias_table_name}.{clustering_col} AS DATE) in ({joined_string})"
+    # Find intervals of contiguous dates, to cut down on the number of filters needed.
+    intervals = []
+    start_date = dates_to_filter[0]
+    end_date = dates_to_filter[0]
+
+    for i in range(1, len(dates_to_filter)):
+        if dates_to_filter[i] == end_date + timedelta(days=1):
+            end_date = dates_to_filter[i]
+        else:
+            intervals.append((start_date, end_date))
+            start_date = dates_to_filter[i]
+            end_date = dates_to_filter[i]
+    intervals.append((start_date, end_date))
+
+    # For each interval, we create the filter for our clustering column.
+    filters = []
+    for interval in intervals:
+        if interval[0] == interval[1]:
+            filters.append(
+                f"CAST({current_alias_table_name}.{clustering_col} AS DATE) == '{interval[0].strftime('%Y-%m-%d')}'"
+            )
+        else:
+            filters.append(
+                f"(CAST({current_alias_table_name}.{clustering_col} AS DATE) >= '{interval[0].strftime('%Y-%m-%d')}' AND CAST({current_alias_table_name}.{clustering_col} AS DATE) <= '{interval[1].strftime('%Y-%m-%d')}')"
```

Comment:
Sure it should be < and = to, and not just <?
```suggestion
                f"(CAST({current_alias_table_name}.{clustering_col} AS DATE) >= '{interval[0].strftime('%Y-%m-%d')}' AND CAST({current_alias_table_name}.{clustering_col} AS DATE) < '{interval[1].strftime('%Y-%m-%d')}')"
```

Reply:
Each interval made by the code above is inclusive, so the filters should be inclusive too. For a single date for instance, it would be a tuple of ("2025-01-01", "2025-01-01") which < would remove which is not intended.